frontend/src/concepts/pipelines/topology/parseUtils.ts

```diff
@@ -159,24 +159,131 @@ export const parseRuntimeInfo = (
     startTime: thisTaskDetail[0].start_time,
     completeTime: thisTaskDetail[0].end_time,
     state: lowestProgress(thisTaskDetail),
-    taskId: thisTaskDetail[0].task_id,
+    taskId: `task.${taskId}`,
     podName: thisTaskDetail[0].child_tasks?.find((o) => o.pod_name)?.pod_name,
   };
 };
 
-export const translateStatusForNode = (stateKF?: RuntimeStateKF): RunStatus | undefined => {
-  switch (stateKF) {
+export const parseRuntimeInfoFromExecutions = (
+  taskId: string,
+  executions: Execution[] | null,
+): PipelineTaskRunStatus | undefined => {
+  if (!executions) {
+    return undefined;
+  }
+
+  const execution = executions.find(
+    (e) => e.getCustomPropertiesMap().get('task_name')?.getStringValue() === taskId,
+  );
+
+  if (!execution) {
+    return undefined;
+  }
+
+  const lastUpdatedTime = execution.getLastUpdateTimeSinceEpoch();
+  let completeTime;
+  if (
+    lastUpdatedTime &&
+    (execution.getLastKnownState() === Execution.State.COMPLETE ||
+      execution.getLastKnownState() === Execution.State.FAILED ||
+      execution.getLastKnownState() === Execution.State.CACHED ||
+      execution.getLastKnownState() === Execution.State.CANCELED)
+  ) {
+    completeTime = new Date(lastUpdatedTime).toISOString();
+  }
+  return {
+    startTime: new Date(execution.getCreateTimeSinceEpoch()).toISOString(),
+    completeTime,
+    state: getResourceStateText({
+      resourceType: ResourceType.EXECUTION,
+      resource: execution,
+      typeName: 'Execution',
+    }),
+    taskId: `task.${taskId}`,
+    podName: execution.getCustomPropertiesMap().get('pod_name')?.getStringValue(),
+  };
+};
+
+export enum ResourceType {
+  ARTIFACT = 'ARTIFACT',
+  EXECUTION = 'EXECUTION',
+}
+
+export interface ArtifactProps {
+  resourceType: ResourceType.ARTIFACT;
+  resource: Artifact;
+  typeName: string;
+}
+
+export interface ExecutionProps {
+  resourceType: ResourceType.EXECUTION;
+  resource: Execution;
+  typeName: string;
+}
+
+export type ResourceInfoProps = ArtifactProps | ExecutionProps;
+
+// Get text representation of resource state.
+// Works for both artifact and execution.
+export const getResourceStateText = (props: ResourceInfoProps): string | undefined => {
```

Comment:
At one point recently all parseUtils were unit tested and the file had 100% line coverage. I think it best we continue to unit test these functions since there was a ticket dedicated to adding tests for this file I assume for good reason.